File: docs/test/samples/csharp/UtilityLibraries/StringLibrary/Class1.cs
using System;

namespace UtilityLibraries
{
    public static class StringLibrary
    {
        public static bool StartsWithUpper(this string s)
        {
            if (String.IsNullOrWhiteSpace(s))
                return false;

            return Char.IsUpper(s[0]);
        }

        public static bool StartsWithLower(this string s)
        {
            if (String.IsNullOrWhiteSpace(s))
                return false;

            return Char.IsLower(s[0]);
        }

        public static bool HasEmbeddedSpaces(this string s)
        {
            foreach (var ch in s.Trim())
            {
                if (ch == ' ')
                    return true;
            }
            return false;
        }
    }
}

File: docs/test/samples/visual-basic/UtilityLibraries/StringLibrary/Class1.vb

Imports System.Runtime.CompilerServices

Namespace UtilityLibraries
    Public Module StringLibrary
        <Extension>
        Public Function StartsWithUpper(s As String) As Boolean
            If (String.IsNullOrWhiteSpace(s)) Then
                Return False
            End If

            Return Char.IsUpper(s(0))
        End Function

        <Extension>
        Public Function StartsWithLower(s As String) As Boolean
            If (String.IsNullOrWhiteSpace(s)) Then
                Return False
            End If

            Return Char.IsLower(s(0))
        End Function

        <Extension>
        Public Function HasEmbeddedSpaces(s As String) As Boolean
            For Each ch In s.Trim()
                If ch = " "c Then
                    Return True
                End If
            Next

            Return False
        End Function
    End Module
End Namespace

